pkg/azure/aks/pricing_map.go

```diff
@@ -0,0 +1,187 @@
+package aks
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+	retailPriceSdk "gomodules.xyz/azure-retail-prices-sdk-for-go/sdk"
+)
+
+const (
+	AZ_API_VERSION string = "2023-01-01-preview" // using latest API Version https://learn.microsoft.com/en-us/rest/api/cost-management/retail-prices/azure-retail-prices
+)
+
+type MachineOperatingSystem int
+
+const (
+	Linux MachineOperatingSystem = iota
+	Windows
+)
+
+var machineOperatingSystemNames [2]string = [2]string{"Linux", "Windows"}
+
+func (o MachineOperatingSystem) String() string {
+	return machineOperatingSystemNames[o-1]
+}
+
+type MachinePriority int
+
+const (
+	OnDemand MachinePriority = iota
+	Spot
+)
+
+var machinePriorityNames [2]string = [2]string{"OnDemand", "Spot"}
+
+func (v MachinePriority) String() string {
+	return machinePriorityNames[v-1]
+}
+
+type PriceBySku map[string]retailPriceSdk.ResourceSKU
+
+type PriceByOperatingSystem map[MachineOperatingSystem]PriceBySku
+
+type PriceByPriority map[MachinePriority]PriceByOperatingSystem
+
+type PriceStore struct {
+	lock              *sync.RWMutex
+	subscriptionId    string
+	logger            *slog.Logger
+	context           context.Context
+	retailPriceClient *retailPriceSdk.RetailPricesClient
+
+	RegionMap map[string]PriceByPriority
+	Cache     map[string]*retailPriceSdk.ResourceSKU
+}
+
+func NewPricingStore(subId string, priceClient *retailPriceSdk.RetailPricesClient, parentLogger *slog.Logger, parentContext context.Context) *PriceStore {
+	logger := parentLogger.With("subsystem", "pricingMap")
+
+	p := &PriceStore{
+		lock:              &sync.RWMutex{},
+		logger:            logger,
+		context:           parentContext,
+		subscriptionId:    subId,
+		retailPriceClient: priceClient,
+
+		RegionMap: make(map[string]PriceByPriority),
+		Cache:     make(map[string]*retailPriceSdk.ResourceSKU),
+	}
+
+	go func() {
+		err := p.PopulatePriceStore([]string{})
+		if err != nil {
+			p.logger.LogAttrs(p.context, slog.LevelError, "error populating initial price store", slog.String("error", err.Error()))
+		}
+	}()
+
+	return p
+}
+
+func (p *PriceStore) buildQueryFilter(locationList []string) string {
```

Comment:
After seeing the usage via tests, it's pretty clear this likely doesn't need to be a method of `PriceStore` and would be better off having it as a private method for the `aks` module.